test/e2e/exec_test.go

```diff
@@ -548,4 +549,24 @@ RUN useradd -u 1000 auser`, fedoraMinimal)
 		session.WaitWithDefaultTimeout()
 		Expect(session.OutputToString()).To(Not(ContainSubstring(secretsString)))
 	})
+
+	It("podman exec --wait 2 seconds on bogus container", func() {
+		errString := "Error: timed out waiting for container:"
+		err := errors.New(errString)
+		session := podmanTest.Podman([]string{"exec", "--wait", "2", "1234"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(125))
+		Expect(err.Error()).To(ContainSubstring(errString))
+	})
+
+	It("podman exec --wait for started container", func() {
+		session := podmanTest.Podman([]string{"run", "-d", "--name", "waitCtr", ALPINE, "top"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		session = podmanTest.Podman([]string{"exec", "--wait", "5", "waitCtr", "whoami"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
```

Comment:
Because this is actually a very important use case of the wait for your quadlet container create race condition.

move `session = podmanTest.Podman([]string{"exec", "--wait", "5", "waitCtr", "whoami"})` before the run command but leave the `WaitWithDefaultTimeout()` and the `Exit()` check after the podman run. That way we should guarantee that wait will function as expected. I also suggest you change the `WaitWithDefaultTimeout()` to `Wait(6)`, the timeout should really take effect end cause an exit. If it this command ever runs longer then 6 seconds we should consider that a fail.

Reply:
great suggestions. Thank you! Updating PR..